File: specification/specs/modules/snapshot_and_restore/repositories/create_repository/CreateRepositoryRequest.ts
@rest_spec_name('snapshot.create_repository')
@class_serializer('CreateRepositoryFormatter')
class CreateRepositoryRequest extends RequestBase {
  path_parts?: {
    repository: Name
  }
  query_parameters?: {
    master_timeout?: Time
    timeout?: Time
    verify?: boolean
  }
  body?: {
    repository?: SnapshotRepository
  }
}

File: output/csharp/Modules/SnapshotAndRestore/Repositories/CreateRepository/CreateRepositoryRequest.cs
using Nest.CommonOptions;
using Nest.Modules;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Modules {

	public class CreateRepositoryRequest : RequestBase {
		
		[DataMember(Name="master_timeout")]
		public Time MasterTimeout { get; set; }


		[DataMember(Name="timeout")]
		public Time Timeout { get; set; }


		[DataMember(Name="verify")]
		public bool Verify { get; set; }


		[DataMember(Name="repository")]
		public SnapshotRepository Repository { get; set; }

	}
}
